File: appointments-app-backend/src/domain/appointment.ts
import { Patient } from './patient';

export class Appointment {
    constructor(
        public id: string,
        public status: string,
        public subject: Patient,
        public start: string,
        public end: string,
    ) {}

    hasSubject(id: string): boolean {
        return this.subject.id === id;
    }
}

File: appointments-app-backend-python-pip/src/domain/appointment.py
from src.domain.patient import Patient

class Appointment:
    def __init__(self, id: str, status: str, subject: Patient, start: str, end: str):
        self.id = id
        self.status = status
        self.subject = subject
        self.start = start
        self.end = end

    def to_dict(self):
        return {
            "id": self.id,
            "status": self.status,
            "subject": self.subject.to_dict(),
            "start": self.start,
            "end": self.end,
        }
